### ISSUE TITLE
Load: Test updated load function

### ISSUE THREAD

**Issue by schin99** [uchicago-cs/chiventure#223]
Once load.c is updated with game-states new function and their functions that previously were only in their documentation is pushed to master, test the load function and make sure our code works and game-state's functions work with our code.

**Comment by jonahdf** [comment]
I got two more tests to pass:
-Before it failed a long desc test, which was strange to me because it passed the other ones. I figured out this was just due to a spelling mistake ;)
-Loading health (and xp, level) didn't work before because the get_foo functions needed to be updated manually. However, I determined these could just be required fields in protobufs, since game state's new_player function automatically loads in a value for all of them (there will never be NULL or equivalent). I made the change in the proto file and load in dev, and then merged to both testing branches since this change affects both load and save.

While there is only 1 failing test right now, we should probably design more to reflect the more complex attribute structs. Also the 1 failing test is in load_game, so it will probably take longer to figure out. I also want to try testing with a WDL loaded game so we can see if the correct members are overwritten.

**Comment by chrisfernandes18** [comment]
All the load tests pass now. We just need to add more detail to the tests to make sure every aspect works correctly, such as testing length of the hash table and successful value passing. So now onto making more details.

**Comment by jonahdf** [comment]
We've been testing load by using the demo branch, which is more an integration-style test (we are actually calling load). We are getting lots of seg faults, and working out each issue one-by-one. We also edited the tests a bit in our most recent PR to master so it did different tests in load_player and load_room (before, it was just doing the same load_item tests in both)

**Comment by jonahdf** [comment]
By creating/testing our demo, we found issues with pointers, deleting items, and much more. We have now worked all of this out, and the demo successfully loads the entire game (including changes in room, item, etc.) We also edited some of the criterion assertions to reflect the tests intended, and all 8 tests pass. It has all been merged into checkpointing/dev, which is included in our final PR #311.

Good luck to everyone in the future who uses our previous work in checkpointing!
